src/vm/eventtrace.cpp

```diff
@@ -7347,6 +7403,140 @@ VOID ETW::EnumerationLog::EnumerationHelper(Module *moduleFilter, BaseDomain *do
     }
 }
 
+bool ETW::TieredCompilationLog::Runtime::IsEnabled()
+{
+    CONTRACTL {
+        NOTHROW;
+        GC_NOTRIGGER;
+    } CONTRACTL_END;
+
+    return
+        ETW_TRACING_CATEGORY_ENABLED(
+            MICROSOFT_WINDOWS_DOTNETRUNTIME_PROVIDER_Context,
+            TRACE_LEVEL_INFORMATION,
+            CLR_TIERED_COMPILATION_KEYWORD);
+}
+
+bool ETW::TieredCompilationLog::Rundown::IsEnabled()
+{
+    CONTRACTL {
+        NOTHROW;
+        GC_NOTRIGGER;
+    } CONTRACTL_END;
+
+    return
+        ETW_TRACING_CATEGORY_ENABLED(
+            MICROSOFT_WINDOWS_DOTNETRUNTIME_RUNDOWN_PROVIDER_Context,
+            TRACE_LEVEL_INFORMATION,
+            CLR_TIERED_COMPILATION_RUNDOWN_KEYWORD);
+}
+
+void ETW::TieredCompilationLog::GetSettings(UINT32 *flagsRef)
+{
+    CONTRACTL {
+        NOTHROW;
+        GC_NOTRIGGER;
+    } CONTRACTL_END;
+    _ASSERTE(g_pConfig->TieredCompilation());
+    _ASSERTE(flagsRef != nullptr);
+
+    enum class Flags : UINT32
```

Comment:
I'd recommend also having a bit for R2R method tiering so that if we ever had a mode that disabled this it would be easy to know this.

Reply:
Currently this event is only sent when tiered compilation is enabled. I was thinking along the lines of potentially having a generic `RuntimeConfig` event that could have all sorts of config options for the runtime. Do you think that would be useful going forward? If so I'll go ahead and change this event to be a generic event and add flags for whether TC is enabled, and whether R2R is enabled if you think that would be useful. There is also the `RuntimeInformation` event that we could piggyback on by adding a new version instead of having another event, but I'm leaning towards having a separate event for config as `RuntimeInformation` then may end up being too generic and may eventually expand to contain too many different kinds of info.

Reply:
I'm still leaning towards having feature-specific events for config. There is already precedence with `GC/Settings` and I think that makes sense, as the event is categorized and you would only get config pieces for categories that are enabled.

Reply:
I think a feature-specific event is fine.  We have some precedent for both, but really it doesn't hurt and there aren't enough for it to really be an issue.  I would lean towards feature-specific things if the feature itself is likely to have more config in the future.

Reply:
Yea, if we happen to learn that certain config options are very useful outside of feature-specific events we could certainly add a generic event in the future to convey that info redundantly. Generic events have the con that they have to last, so that would probably be a hard sell anyway.